src/OpenTelemetry/Internal/TagTransformer.cs

```diff
@@ -33,7 +34,7 @@ public bool TryTransformTag(KeyValuePair<string, object> tag, out T result)
         {
             case char:
             case string:
-                result = this.TransformStringTag(tag.Key, Convert.ToString(tag.Value));
+                result = this.TransformStringTag(tag.Key, TruncateString(Convert.ToString(tag.Value), maxLength));
```

Comment:
Do we need to do this for the `default` case as well?

Reply:
Yes, good catch thanks. Fixed and added a test.